### ISSUE TITLE
Is it possible to pass multiple data generators for multiple inputs to model.fit_generator function

### ISSUE THREAD

**Issue by s1782662** [keras-team/keras#9613]
Model accepts three inputs. I am using keras data generator. Is it possible to pass three data generator to model.fit_generator function ?. 
 

Problem Definition
------------------
I am working a classification problem. The dataset i am using is Painters by number, a competition hosted by kaggle . The task is to identify painter,style and genre given paintings.

I have developed individual models to perform each task. Now, i would like to incorporate multi task learning, see if it outperforms individual models. 

------                                             ---------------
Model            ----------------                                 No of classes (Softmax)
------                                                  ---------------
Model predicting painter  ---------------          8
given paintings

Model predicting style     ---------------           10
given paintings      

Model predicting genre   ---------------            23
given paintings    

------                     ---------------

Now, i want to do multi task learning , so I came up with below simple architecture

style   = Input(shape=(64,64,3))
genre   = Input(shape=(64,64,3))
painter = Input(shape=(64,64,3))

shared_conv = Convolution2D(
                filters = 5,# 5 feature maps
                kernel_size = (5,5),
                strides = 1) 

shared_conv_layer_A = shared_conv(style)
shared_conv_layer_B = shared_conv(genre)
shared_conv_layer_C = shared_conv(painter)

merged_layer = keras.layers.concatenate([shared_conv_layer_A,shared_conv_layer_B,shared_conv_layer_C],axis=-1)

pooling = MaxPooling2D(
            pool_size = (2,2),
            strides = 2
          )(merged_layer)

dense = Flatten()(pooling)

out_style = Dense(
            no_classes_style, 
            kernel_initializer=glorot_normal(seed=seed_val), 
            bias_initializer = 'zero', 
            kernel_regularizer = l2(l=0.0001),
            activation = 'softmax',
        )(dense)

out_genre = Dense(
            no_classes_genre, 
            kernel_initializer=glorot_normal(seed=seed_val), 
            bias_initializer = 'zero', 
            kernel_regularizer = l2(l=0.0001),
            activation = 'softmax',
        )(dense)

out_painter = Dense(
            no_classes_painter, 
            kernel_initializer=glorot_normal(seed=seed_val), 
            bias_initializer = 'zero', 
            kernel_regularizer = l2(l=0.0001),
            activation = 'softmax',
        )(dense)

multi_tasking_model = Model(inputs=[style,genre,painter],outputs=[out_style,out_genre,out_painter])
multi_tasking_model.summary()

multi_tasking_model.compile(
    loss = 'categorical_crossentropy', 
    optimizer=Adam(lr=0.0001, beta_1=0.9, beta_2=0.999, epsilon=0.00000001 ),
    metrics=['accuracy'] 
)

**history = multi_tasking_model.fit_generator(
    generator = [trainStyleDataGenerator,trainGenreDataGenerator,trainPainterDataGenerator],
    steps_per_epoch= len(trainStyleDataGenerator),
    epochs = no_epoch,
    validation_data = [validationStyleDataGenerator,validationGenreDataGenerator,validationPainterDataGenerator],
)**

Is there any alternative way. Any suggestions or tips would be greatly helpful please ?.
